MetaMask/metamask-mobile
# test(3615): add new e2e test for initial dapp connection and non permitted flow

## **Description**
Covers with e2e the scenarios where the user connects an in app browser dapp by granting permissions to one account and one chain, aka initial connection. A second e2e test covers one flow of the none permitted flow.




## **Related issues**

Contributes to solve issue: https://github.com/MetaMask/MetaMask-planning/issues/2796

## **Manual testing steps**

I have added screenshot below to see how it looks.

1. `yarn watch:clean` 
2. `yarn test:e2e:ios:debug:build`
3. `yarn test:e2e:ios:debug:run e2e/specs/multichain/permission-system-initial-permissions-grant-one-account-and-one-chain.spec.js`
4. For the second scenario, non permitted flow, not shown in the screenshot below, it is `permission-system-connect-to-non-permitted-chain.spec`


## **Screenshots/Recordings**

![2024-11-28 16 47 48](https://github.com/user-attachments/assets/f9dca549-42fe-4fc2-b2f1-6a8339dcba72)




## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
